lib/rules/prop-types.js

```diff
@@ -967,7 +942,7 @@ module.exports = {
         const directDestructuring =
           destructuring &&
           PROPS_REGEX.test(node.init.name) &&
-          (utils.getParentStatelessComponent() || inConstructor() || inComponentWillReceiveProps())
```

Comment:
also, which test cases warrant this change from just checking willReceiveProps, to checking all the lifecycle methods?

Reply:
This is missing tests! Indeed we will now check more cases than just `componentWillReceiveProps`